raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -996,19 +1001,27 @@ def mediate_transfer(
     send a refund back to the payer, allowing the payer to try a different
     route.
     """
-    reachable_routes = filter_reachable_routes(possible_routes, nodeaddresses_to_networkstates)
-    available_routes = filter_used_routes(state.transfers_pair, reachable_routes)
-
     assert payer_channel.partner_state.address == payer_transfer.balance_proof.sender
 
-    transfer_pair, mediated_events = forward_transfer_pair(
-        payer_transfer,
-        available_routes,
-        channelidentifiers_to_channels,
-        pseudo_random_generator,
-        block_number,
+    transfer_pair: Optional[MediationPairState] = None
+    mediated_events: List[Event] = list()
+
+    reachable_route_states = filter_reachable_routes(
+        candidate_route_states, nodeaddresses_to_networkstates
     )
 
+    for route_state in reachable_route_states:
```

Comment:
Could you please move the route filtering to a single function? It is important to call ` filter_reachable_routes`, `filter_used_routes`, and `channel.is_channel_usable`, having them together should it make it easier to review.

Reply:
I am not sure I follow. The idea would be to have one "master" function that takes all the logic related to these functions? At the moment, the check for "is_channel_usable" is only at the mediator part.